src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs

```diff
@@ -102,14 +102,14 @@ public int GetJobCount(string[] tags, string stateName = null)
                     $@")
 select top {maxTags} j.StateName AS [Key], count(*) AS [Value]
 from [{_options.SchemaName}].Job j with (nolock)
-inner join cte on cte.Id = j.Id 
-inner join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id
+inner join cte on cte.Id = j.Id
+inner join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id and j.id = s.jobId
```

Comment:
Index on State table contains both Id and JobId.
Although this looks redundant, it reduces query cost from 950 to 0.5 with millions jobs in the table